lib/services/query/selectPopulatedFields.js

```diff
@@ -13,7 +13,8 @@ module.exports = function selectPopulatedFields(query) {
     var userProvidedFields = query._userProvidedFields || {};
     if (query.selectedInclusively()) {
       for (i = 0; i < paths.length; ++i) {
-        if (!isPathInFields(userProvidedFields, paths[i])) {
+        var hasPath = query._fields[paths[i]];
```

Comment:
I'm suspicious of this. For one thing, `query._fields[paths[i]]` may be `0`, `false`, etc. so checking for falsy will catch both cases where both the field isn't in the projection and if the field is explicitly excluded from the projection.

For another, I'm not so sure that #6546 is a bug. Let's discuss that more.

Reply:
Yeah, I wasn't sure either. 

The idea of calling populate on a path that isn't selected seems like a weird thing to want to do. I played mental tennis for a while with the idea, eventually I just assumed the requester was passing in a select string as a variable and always calling populate with the same options. This would allow you to make a complex populate object that you only generate once but you can alter it's behavior with a simple string in the select.

I thought about pushing back that the populate should only be called on selected paths, and that it could be achieved fairly easily by generating the populate dynamically as well as the select string. The only reason why I didn't was because the change they pointed out in the bug did alter the behavior ( i tested it explicitly on the 4.x version just before the change).